src/fastgltf_tools.hpp

```diff
@@ -0,0 +1,467 @@
+		}
+
+		auto nextSparseIndex = internal::getAccessorElementAt<std::uint32_t>(
+				accessor.sparse->indexComponentType, indicesBytes);
+		std::size_t sparseIndexCount = 0;
+
+		for (std::size_t i = 0; i < accessor.count; ++i) {
+			if (i == nextSparseIndex) {
+				func(internal::getAccessorElementAt<ElementType>(accessor.componentType,
+						valuesBytes + valueStride * sparseIndexCount));
+
+				++sparseIndexCount;
+
+				if (sparseIndexCount < accessor.sparse->count) {
+					nextSparseIndex = internal::getAccessorElementAt<std::uint32_t>(
+							accessor.sparse->indexComponentType, indicesBytes + indexStride * sparseIndexCount);
+				}
+			} else if (accessor.bufferViewIndex) {
+				func(internal::getAccessorElementAt<ElementType>(accessor.componentType,
+						srcBytes + srcStride * i));
+			} else {
+				func(ElementType{});
+			}
+		}
+	} else {
+		// 5.1.1. accessor.bufferView
+		// The index of the buffer view. When undefined, the accessor MUST be initialized with zeros; sparse
+		// property or extensions MAY override zeros with actual values.
+		if (!accessor.bufferViewIndex) {
+			for (std::size_t i = 0; i < accessor.count; ++i) {
+				func(ElementType{});
+			}
+		}
+		else {
+			auto& view = asset.bufferViews[*accessor.bufferViewIndex];
+			auto stride = view.byteStride ? *view.byteStride : getElementByteSize(accessor.type, accessor.componentType);
+
+			auto* bytes = adapter(asset.buffers[view.bufferIndex]);
+			bytes += view.byteOffset + accessor.byteOffset;
+
+			for (std::size_t i = 0; i < accessor.count; ++i) {
+				func(internal::getAccessorElementAt<ElementType>(accessor.componentType, bytes + i * stride));
+			}
+		}
+	}
+}
+
+template <typename ElementType, std::size_t TargetStride = sizeof(ElementType),
+		 typename BufferDataAdapter = DefaultBufferDataAdapter>
+#if FASTGLTF_HAS_CONCEPTS
+requires Element<ElementType>
+#endif
+void copyFromAccessor(const Asset& asset, const Accessor& accessor, void* dest,
+		const BufferDataAdapter& adapter = {}) {
+	using Traits = ElementTraits<ElementType>;
+	static_assert(Traits::type != AccessorType::Invalid, "Accessor traits must provide a valid Accessor Type");
+	static_assert(std::is_default_constructible_v<ElementType>, "Element type must be default constructible");
+	static_assert(std::is_constructible_v<ElementType>, "Element type must be constructible");
+	static_assert(std::is_move_assignable_v<ElementType>, "Element type must be move-assignable");
+
+	if (accessor.type != Traits::type) {
+		return;
+	}
+
+	auto* dstBytes = reinterpret_cast<std::byte*>(dest);
+
+	if (accessor.sparse && accessor.sparse->count > 0) {
+		std::size_t index = 0;
+		iterateAccessor<ElementType>(asset, accessor, [&](auto&& value) {
+			auto* pDest = reinterpret_cast<ElementType*>(dstBytes + TargetStride * index);
+			*pDest = std::move(value);
+			++index;
+		}, adapter);
+	} else {
+		auto elemSize = getElementByteSize(accessor.type, accessor.componentType);
```

Comment:
This should probably not be nested. I think you should return after the iterateAccessor and then remove the else, like you do a little further down.

Reply:
Fixed in [df17edc](https://github.com/spnda/fastgltf/pull/20/commits/df17edc742f1893ece80c0cc2467a9b68b7ec880)